python/sidecar/http.py

```diff
@@ -230,10 +226,10 @@ async def delete_project(project_id: str):
 
 
 @project_api.get("/{project_id}/files")
-async def get_project_files(project_id: str):
+async def get_project_files(project_id: str) -> list[FileEntry | FolderEntry]:
     user_id = "user1"
-    filepaths = projects.get_project_files(user_id, project_id)
-    return filepaths
+    response = projects.get_project_files(user_id, project_id)
```

Comment:
IMO this method should only return the filesystem content.

Reply:
that's what's happening on the line below

Reply:
I see that. My concern is why do you call `projects.get_project_files(user_id, project_id)` if you don't need the files. That's extra work (reading the filesystem) for nothing. I think you should create methods to split that method (of projects) in two (one to read project info and the other for files)